src/main/java/fr/sncf/osrd/train/TrainPositionTracker.java

```diff
@@ -180,16 +180,23 @@ public double getPathPosition() {
         return pathPosition;
     }
 
-    /** Computes the maximum grade (slope) under the train. */
-    public double maxTrainGrade() {
-        var val = 0.;
+    /** Computes the average grade (slope) under the train. */
+    public double meanTrainGrade() {
```

Comment:
This function should weight the gradient under the train according to the portion of the train under each gradient. To do so the function `getValuesInRange` could return the range and the value of each gradient